primitives/transaction/tests/htlc_contract_verify.rs

```diff
@@ -128,70 +127,63 @@ fn it_can_verify_regular_transfer() {
 
     // regular: valid Blake-2b
     let proof = OutgoingHTLCTransactionProof::RegularTransfer {
-        hash_algorithm: HashAlgorithm::Blake2b,
         hash_depth: 1,
-        hash_root: AnyHash::from(<[u8; 32]>::from(
-            Blake2bHasher::default().digest(&[0u8; 32]),
-        )),
-        pre_image: AnyHash::from([0u8; 32]),
+        hash_root: AnyHash::from(Blake2bHasher::default().digest(&[0u8; 32])),
+        pre_image: AnyHash::Blake2b(AnyHash32::from([0u8; 32])),
```

Comment:
Technically, the preimage is not necessarily a hash of any kind. It's just random data, not even technically a _hash_ at all. A secret basically.

Reply:
Yeah, this is not elegant. One solution could be to use another type for the `pre_image` instead of `AnyHash`.